### DISCUSSION TITLE
Add node-snap7 to the default Docker installation

### DISCUSSION THREAD

**Discussion by mauroalexandre** [frangoteam/FUXA#1265] [Polls]
I cannot provide the number of users utilizing Siemens PLCs via S7, but I would like to propose that the installation of node-snap7 be included in the Dockerfile by default.

**Comment by unocelli** [comment]
I propose to create a docker version (tagged s7) with node-snap7

**Reply by mauroalexandre** [reply]
Also, I believe it to be a good option.

Another option is to create an `ARG` in the Dockerfile like this:

```dockerfile
ARG NODE_SNAP=false

RUN if [ "$NODE_SNAP" = "true" ]; then \
    npm install node-snap7; \
    fi
```

And build the Docker image with the input argument `NODE_SNAP` like this:

```sh
docker build --build-arg NODE_SNAP=true ...
```

**Comment by unocelli** [comment]
Thanks @mauroalexandre 
I created the docker version with installed node-snap7 library in frangoteam/fuxa:snap7
